### PR TITLE
TSI Fixes (1.3)

### PR DESCRIPTION
**Pull request by benbjohnson**
Cherry picking for 1.3:

* https://github.com/influxdata/influxdb/pull/8475
* https://github.com/influxdata/influxdb/pull/8468

###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated

### CODE REVIEW

File: tsdb/index/tsi1/file_set.go [hunk 1]
```diff
@@ -78,8 +78,9 @@ func (fs *FileSet) MustReplace(oldFiles []File, newFile File) *FileSet {
 
 	// Ensure all old files are contiguous.
 	for j := range oldFiles {
+		println("dbg/replace", len(fs.files), "//", i, j)
```
Comment (jwilder) [inline tsdb/index/tsi1/file_set.go]: Remove?

File: tsdb/index/tsi1/file_set.go [hunk 2]
```diff
@@ -78,8 +78,9 @@ func (fs *FileSet) MustReplace(oldFiles []File, newFile File) *FileSet {
...
 		if fs.files[i+j] != oldFiles[j] {
-			panic("cannot replace non-contiguous files")
+			panic(fmt.Sprintf("cannot replace non-contiguous files: subset=%+v, fileset=%+v", Files(oldFiles).IDs(), Files(fs.files).IDs()))
```
Comment (jwilder) [inline tsdb/index/tsi1/file_set.go]: Is `MustReplace` only used in tests?

File: tsdb/index/tsi1/file_set.go (hunk 1 above)
Comment (benbjohnson) [inline tsdb/index/tsi1/file_set.go]: Fixed in 400147c.

File: tsdb/index/tsi1/file_set.go (hunk 2 above)
Comment (benbjohnson) [inline tsdb/index/tsi1/file_set.go]: No, this is called in a goroutine started by `compact()`. It's an invariant of the code and should not fail. It's an extra check to ensure the index does not get corrupted.

**Pull request merged**